File: 0036-valid-sudoku/0036-valid-sudoku.java
import java.util.*;

class Solution {
    public boolean isValidSudoku(char[][] board) {

        // validate rows
        for (int i = 0; i < 9; i++) {
            Set<Character> set = new HashSet<>();
            for (int j = 0; j < 9; j++) {
                char item = board[i][j];
                if (item != '.') {
                    if (set.contains(item)) {
                        return false;
                    }
                    set.add(item);
                }
            }
        }
        // validate columns
        for (int i = 0; i < 9; i++) {
            Set<Character> set = new HashSet<>();

            for (int j = 0; j < 9; j++) {
                char item = board[j][i];

                if (item != '.') {
                    if (set.contains(item)) {
                        return false;
                    }
                    set.add(item);
                }
            }
        }

        // validate 3x3 boxes
        for (int i = 0; i < 9; i += 3) {
            for (int j = 0; j < 9; j += 3) {
                Set<Character> set = new HashSet<>();
                for (int row = i; row < i + 3; row++) {
                    for (int col = j; col < j + 3; col++) {
                        char item = board[row][col];
                        if (item != '.') {
                            if (set.contains(item)) {
                                return false;
                            }
                            set.add(item);
                        }
                    }
                }
            }
        }

        return true;
    }
}

File: 0036-valid-sudoku/0036-valid-sudoku.py
class Solution:
    def isValidSudoku(self, board: List[List[str]]) -> bool:
        #rows
        for i in range (9):
            seen = set()
            for j in range (9):
                if board[i][j] != ".":
                    if board[i][j] in seen :
                        return False
                    seen.add(board[i][j])

        for i in range (9):
            seen = set()
            for j in range (9):
                if board[j][i] != ".":
                    if board[j][i] in seen :
                        return False
                    seen.add(board[j][i])

        for i in range(0,9,3):
            for j in range(0,9,3):
                seen = set()
                for row in range(i,i+3):
                    for col in range(j, j+3):
                        if board[row][col] != ".":
                            if board[row][col] in seen :
                                return False
                            seen.add(board[row][col])
        return True

